### PR TITLE
Make expectCrash(executing:_) safer

### PR DESCRIPTION
**Pull request by frootloops**
Check out https://github.com/apple/swift/pull/13411 for details

### CODE REVIEW

**Comment by jrose-apple** [comment]
I like this version!

File: stdlib/private/StdlibUnittest/StdlibUnittest.swift.gyb [hunk 1]
```diff
@@ -542,9 +542,11 @@ public func expectCrashLater() {
   _seenExpectCrash = true
 }
 
-public func expectCrash(executing: () -> Void) {
+public func expectCrash(executing: () -> Void) -> Never {
   expectCrashLater()
   executing()
+  expectUnreachable()
+  fatalError()
```
Comment (moiseev) [inline stdlib/private/StdlibUnittest/StdlibUnittest.swift.gyb]: This one is unnecessary ;-) `expectUnreachable` already terminates execution.

**Review by moiseev — APPROVED** [review]

**Comment by moiseev** [comment]
@swift-ci Please smoke test and merge

File: stdlib/private/StdlibUnittest/StdlibUnittest.swift.gyb (hunk 1 above)
Comment (jrose-apple) [reply stdlib/private/StdlibUnittest/StdlibUnittest.swift.gyb]: It *would* be unnecessary if `expectUnreachable` returned `Never`…

**Pull request merged**